src/Features/CSharp/Portable/Completion/CompletionProviders/NamedParameterCompletionProvider.cs

```diff
@@ -258,7 +258,8 @@ int IEqualityComparer<IParameterSymbol>.GetHashCode(IParameterSymbol obj)
                 selectedItem.Span,
                 // Do not insert colon on <Tab> so that user can complete out a variable name that does not currently exist.
                 // ch == null is to support the old completion only.
-                (ch == null || ch == '\t') ? selectedItem.DisplayText : selectedItem.GetEntireDisplayText()));
+                // Do not insert an extra colon if colon has been explicitly typed.
+                (ch == null || ch == '\t' || ch == ':') ? selectedItem.DisplayText : selectedItem.GetEntireDisplayText()));
```

Comment:
was this a regression?  i just tried this with the existing roslyn, and things seem to work properly.  did something change here?

Reply:
yes , this was a regression ..introduced by https://github.com/dotnet/roslyn/pull/31509/files#diff-6c64e8d5ccf958759fb01a0c663b4246